### DISCUSSION TITLE
Aprori and FP growth shows wrong itemsets

### DISCUSSION THREAD

**Discussion by jahidhasanlinix** [rasbt/mlxtend#868] [Q&A]
I was following the mlxtend docs to do the experiment on Apriori and FP growth and compare it but it seems I am getting different itemsets where it should be the same as I use the same df. Any suggestions why this occurring?

**Comment by rasbt** [comment]
Thanks for sharing your concern. One thing we could look at is if the arrays maybe just have a different ordering. Could you try comparing the sorted DataFrames? E.g., by 

```python
apriori_itemsets.sort_values('itemsets')[:5]
```

and

```python
fp_itemsets.sort_values('itemsets')[:5]
```

**Reply by jahidhasanlinix** [reply]
Thank you for your response. So I tried with the **`sort_values()`** It did change the itemsets in the output but both itemsets are not the same, it's still different. Is there any suggestion afterward to fix this issue?

**Comment by rasbt** [comment]
Hm sorry, could you sort by support?

**Reply by jahidhasanlinix** [reply]
Still the same. I am not sure why this, it should be the same as I followed the documentation correctly... Its like they have same support, but itemsets are different in both cases.

**Comment by jahidhasanlinix** [comment]
I tried like this, this gives some matched itemsets but still not accurate as it gives different index numbers even some matches in both cases.
`**apriori_itemsets.sort_values(['support','itemsets'], ascending=[True, False])[:5]**`

**Reply by rasbt** [reply]
Hm, I see. I would say the index numbers are not really relevant as they algorithms process the itemsets in a different order. Not sure what's going on to be honest. Would it be possible to increase the support so that the resulting dataframe is a bit smaller so that we can manually take a look at the dataframes from both algorithms?

**Reply by jahidhasanlinix** [reply]
Seems like that does not work as well. Can't figure out yet, how to resolve this issue. As in comparison of both cases, it should be printed same itemsets, but it doesn't.

**Reply by rasbt** [reply]
yeah it should give the same itemsets. I am not sure what's going on either.

**Reply by jahidhasanlinix** [reply]
So finally I able to resolved it.  **`.sort_values(by=['support','itemsets'], ascending=False)`** made this changes helped me to get same itemsets. Thank you for your suggestions, it really helped me to solve this.

**Reply by rasbt** [reply]
Oh glad to hear! Sorry to hear about the frustration behind it, but it's nice to know that you solved it (and that there's no bug :)). 

That's actually super useful, too! I am thinking whether the sorting should maybe be part of the apriori & fpgrowth functions (or at least mentioned in the docs). I will add it to the issue tracker.

**Reply by jahidhasanlinix** [reply]
Thank you for your response. Yeah, I guess it would be nice if the sorting is a part of of these algorithms.

**Reply by rasbt** [reply]
Sounds good! I will try to add it soon once the semester calms down a bit :)
